common/functions/src/scalars/maths/trigonometric.rs

```diff
@@ -60,29 +69,83 @@ impl Function for TrigonometricFunction {
-        Ok(DataType::Float64)
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        match self.t {
+            Trigonometric::ATAN => Some((1, 2)),
+            _ => None,
+        }
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if is_numeric(&args[0]) || args[0] == DataType::String || args[0] == DataType::Null {
+            Ok(DataType::Float64)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected numeric, but got {}",
+                args[0]
+            )))
+        }
     }
 
     fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
-        Ok(false)
+        Ok(true)
     }
 
     fn eval(&self, columns: &DataColumnsWithField, _input_rows: usize) -> Result<DataColumn> {
-        let result = columns[0]
-            .column()
-            .to_minimal_array()?
-            .cast_with_type(&DataType::Float64)?
-            .f64()?
-            .apply_cast_numeric(|v| match self.t {
-                Trigonometric::COS => v.cos(),
-                Trigonometric::SIN => v.sin(),
-                Trigonometric::COT => 1.0 / v.tan(),
-                Trigonometric::TAN => v.tan(),
+        let result = if columns.len() == 1 {
+            let opt_iter = columns[0]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            let opt_iter = opt_iter.f64()?.into_iter().map(|v| match v {
+                Some(&v) => match self.t {
+                    Trigonometric::COS => Some(v.cos()),
+                    Trigonometric::SIN => Some(v.sin()),
+                    Trigonometric::COT => Some(1.0 / v.tan()),
+                    Trigonometric::TAN => Some(v.tan()),
+                    Trigonometric::ACOS => {
+                        if (-1_f64..=1_f64).contains(&v) {
+                            Some(v.acos())
+                        } else {
+                            None
+                        }
+                    }
+                    Trigonometric::ASIN => {
+                        if (-1_f64..=1_f64).contains(&v) {
+                            Some(v.asin())
+                        } else {
+                            None
+                        }
+                    }
+                    Trigonometric::ATAN => Some(v.atan()),
+                    _ => unreachable!(),
+                },
+                None => None,
             });
+            DFFloat64Array::new_from_opt_iter(opt_iter)
+        } else {
+            let y_series = columns[0]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            let x_series = columns[1]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+
+            let opt_iter = y_series
+                .f64()?
+                .into_no_null_iter()
```

Comment:
x, y may have different sizes.

X: Array[N]   --> to_minimal ---> array[N]
Y: Constant[N] --> to_minimal --> Constant[1]

Reply:
Thanks @sundy-li .

Update [here](https://github.com/datafuselabs/databend/pull/2888/files#diff-e4c1b93add37f16f9ffd647fd6da79377efef26c78eb8b47979636b3145c3c35R136-R163) for check x, y sizes .